src/llvm-multiversioning.cpp

```diff
@@ -697,18 +698,59 @@ Constant *CloneCtx::rewrite_gv_init(const Stack& stack)
     return res;
 }
 
+// replace an alias to a function with a trampoline and (uninitialized) global variable slot
+std::pair<Function *, GlobalVariable *> CloneCtx::rewrite_alias(GlobalAlias *alias)
+{
+    Function *F = cast<Function>(alias->getAliasee());
+
+    auto Name = alias->getName();
+    alias->setName("");
+    Function *trampoline =
+        Function::Create(F->getFunctionType(), alias->getLinkage(), Name, &M);
+    trampoline->copyAttributesFrom(F);
+    alias->eraseFromParent();
+    GlobalVariable *slot =
+        new GlobalVariable(M, F->getType(), false, GlobalVariable::InternalLinkage, NULL,
+                           Name + ".reloc_slot");
+
+    auto BB = BasicBlock::Create(ctx, "top", trampoline);
+    IRBuilder<> irbuilder(BB);
+
+    auto ptr = irbuilder.CreateLoad(F->getType(), slot);
+    ptr->setMetadata(llvm::LLVMContext::MD_tbaa, tbaa_const);
+    ptr->setMetadata(llvm::LLVMContext::MD_invariant_load, MDNode::get(ctx, None));
+
+    std::vector<Value *> Args;
+    for (auto &arg : trampoline->args())
+        Args.push_back(&arg);
+    auto call = irbuilder.CreateCall(ptr, makeArrayRef(Args));
+    call->setTailCallKind(CallInst::TCK_MustTail);
```

Comment:
See the check in `ccall.cpp` regarding vararg function and musttail. It doesn't work everywhere and it should check for vararg functions on unsupported platforms and abort.